selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/NewWorkspacePageTest.java

```diff
@@ -0,0 +1,519 @@
+    seleniumWebDriverHelper.sendKeys(Keys.DELETE.toString());
+    newWorkspace.waitFiltersInputFieldIsEmpty();
+    newWorkspace.deleteLastTagFromInputTagsField();
+
+    newWorkspace.typeToFiltersInput("j");
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA");
+    seleniumWebDriverHelper.sendKeys(Keys.TAB.toString());
+    newWorkspace.waitSelectedFiltersSuggestion("JDK");
+    seleniumWebDriverHelper.sendKeys(Keys.ENTER.toString());
+    newWorkspace.waitFiltersInputFieldTags(asList("JDK"));
+    newWorkspace.clickOnTitlePlaceCoordinate();
+    newWorkspace.waitFiltersFormClosed();
+
+    newWorkspace.getAvailableStacks();
+    newWorkspace.waitStacks(EXPECTED_JDK_STACKS);
+
+    newWorkspace.clickOnFiltersButton();
+    newWorkspace.waitFiltersFormOpened();
+    newWorkspace.waitFiltersInputFieldTags(asList("JDK"));
+    newWorkspace.deleteLastTagFromInputTagsField();
+    newWorkspace.waitFiltersInputFieldIsEmpty();
+    newWorkspace.clickOnTitlePlaceCoordinate();
+    newWorkspace.waitFiltersFormClosed();
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+  }
+
+  @Test
+  public void checkAddStackButton() {
+    newWorkspace.waitPageLoad();
+    newWorkspace.clickOnAddStackButton();
+    newWorkspace.waitCreateStackDialog();
+    seleniumWebDriverHelper.sendKeys(ESCAPE.toString());
+    newWorkspace.waitCreateStackDialogClosing();
+
+    newWorkspace.clickOnAddStackButton();
+    newWorkspace.waitCreateStackDialog();
+    newWorkspace.clickOnTitlePlaceCoordinate();
+    newWorkspace.waitCreateStackDialogClosing();
+
+    newWorkspace.clickOnAddStackButton();
+    newWorkspace.waitCreateStackDialog();
+    newWorkspace.closeCreateStackDialogByCloseButton();
+    newWorkspace.waitCreateStackDialogClosing();
+
+    newWorkspace.clickOnAddStackButton();
+    newWorkspace.waitCreateStackDialog();
+    newWorkspace.clickOnNoButtonInCreateStackDialog();
+    newWorkspace.waitCreateStackDialogClosing();
+
+    newWorkspace.clickOnAddStackButton();
+    newWorkspace.waitCreateStackDialog();
+    newWorkspace.clickOnYesButtonInCreateStackDialog();
+    stacks.waitToolbarTitleName();
+    seleniumWebDriver.navigate().back();
+    newWorkspace.waitPageLoad();
+  }
+
+  @Test
+  public void checkSearchField() {
+    newWorkspace.waitPageLoad();
+    newWorkspace.typeToSearchInput("Java");
+    newWorkspace.waitStacks(EXPECTED_JAVA_STACKS);
+    newWorkspace.typeToSearchInput("");
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+    newWorkspace.typeToSearchInput("java");
+    newWorkspace.waitStacks(EXPECTED_JAVA_STACKS);
+    newWorkspace.typeToSearchInput("");
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+    newWorkspace.typeToSearchInput("JAVA");
+    newWorkspace.waitStacks(EXPECTED_JAVA_STACKS);
+    newWorkspace.typeToSearchInput("");
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+  }
+
+  @Test
+  public void checkRamSelection() {
+    newWorkspace.waitPageLoad();
```

Comment:
There are a lot of different type of steps here and it's hard to understand common logic of test. So, it would be better to have different steps commented, or extracted into the separate tests.